cmd/clusterctl/client/cluster/objectgraph.go

```diff
@@ -133,13 +141,28 @@ func (o *objectGraph) addObj(obj *unstructured.Unstructured) {
 	newNode := o.objToNode(obj)
 
 	// Process OwnerReferences; if the owner object doe not exists yet, create a virtual node as a placeholder for it.
+	o.processOwnerReferences(obj, newNode)
+}
+
+// addRestoredObj adds a Kubernetes object to the object graph that is generated during the restore discovery phase.
+// Populates the restoredObject field to be referenced during restore
+// During add, OwnerReferences are processed in order to create the dependency graph.
+func (o *objectGraph) addRestoredObj(obj *unstructured.Unstructured) {
```

Comment:
Hmmm good point. I'm learning alot about the object graph here! I used your suggestion, but also kept the `processOwnerReferences` code chunk since I couldn't get tests to pass without it. Specifically, the "genericInfrastructure" object would be missing and I guessing that without processing the owner references, the cluster from file won't have the correct reference to objects it owns.

Reply:
this is now addressed